### PR TITLE
Introduce taxonomy for blog pages

### PR DESCRIPTION
**Pull request by abangser**
Hugo has the built in ability to do categories and tags. Using them can enable things like word clouds, pages that highlight aligned content and more.

Right now this is just the first introduction of categories and tags and is being used only for styling a single blog post. This formatting will help us enable more community posts by being clear when things are TAG/WG created vs community submitted.

This will be formalised in #495 and be retrospectively applied to #515 and available for a few more that I know people are interested in proposing.

The proposed taxonomy is:

* *Categories* should be minimal and applicable across a number of tags. Therefore there are only two:
  * announcements
  * community
* *Tags* can help identify sub topics that are more narrow focuses. Right now there are proposed tags:
  * KubeCon
  * Technology
  * WG Multi-tenancy
  * WG Platforms

## Help wanted

I am no designer, to say the least 😅 This could use a look over on they use of Hugo (maybe @cjyabraham ?) and also general desired design (also maybe Chris, but also anyone else!)

### CODE REVIEW

**Comment by cjyabraham** [comment]
This is looking good. Here are a few suggestions for the design:

I think for the tags, we could borrow code and design from [the Glossary site](https://glossary.cncf.io/application-programming-interface/). See how the tags are. You can click on any of them to then [search all content for that tag](https://glossary.cncf.io/tags/?architecture=true):
Screenshot 2023-12-16 at 2 58 30 PM


For the categories, we could put it above the blog title, similar to how we do it on [the CNCF site](https://www.cncf.io/announcements/2023/11/08/cloud-native-computing-foundation-presents-the-top-end-user-award-to-spotify/):

Screenshot 2023-12-16 at 3 00 44 PM

What do you think?

**Comment by abangser** [comment]
Thanks for the quick review @cjyabraham! I think your suggestions make a lot of sense.

Are these simple additions for you? I don't want to leave you with a mess, but also appreciate that you have a lot better handle on this code base than I do!

**Comment by cjyabraham** [comment]
@abangser I think it should be pretty straightforward since I can lift a lot of the code from the Glossary site. I should be able to get to this in the next 3 weeks if you can wait that long?

**Comment by abangser** [comment]
Absolutely. I'd rather it done right than fast and with the holidays that is actually really speedy! Thanks so much for jumping on this 🙇‍♀️

I will reach out to tag leadership to verify the intentions as well before you get started so that we have a smooth merging once done 👍👍

**Comment by cjyabraham** [comment]
Ok, sounds good :)

**Comment by joshgav** [comment]
Thanks for this @abangser! It looks like the tags and categories are arbitrary and anyone can choose any string, is that right? So future posts won't be constrained to an established taxonomy. That begs the question - should we establish an official tag "taxonomy" somehow? That is, so that future writers will have to choose from an established list of terms. Is that possible somehow in Hugo? If not, we can rely on reviews here, and perhaps we should document our tagging patterns.

I think the main question in this PR is if the initial set of categories and tags makes sense. To this - perhaps the two categories should be "announcement" and "article", and "community-contributed" could be a tag on some articles.

File: website/content/en/blog/clusters-for-all-cloud-tenants.md
Comment (joshgav) [inline website/content/en/blog/clusters-for-all-cloud-tenants.md:6]: Perhaps - "Article" and put "Community-contributed" as a tag?

File: website/content/en/blog/clusters-for-all-cloud-tenants.md
Comment (joshgav) [inline website/content/en/blog/clusters-for-all-cloud-tenants.md:9]: I think you're getting at the fact that the article is technical and not some sort of announcement. Perhaps "technical" or "deep dive" would fit better?

**Comment by lianmakesthings** [comment]
Whie we're on the topic of tags. 
@cjyabraham we discussed way back to introduce an "inactive" tag with which we could mark inactive Working groups that we still want to feature ont he site (e.g. operator).
Does it make sense to put that in this PR as well?

File: website/content/en/blog/clusters-for-all-cloud-tenants.md
Comment (abangser) [reply website/content/en/blog/clusters-for-all-cloud-tenants.md:6]: I like this!

**Comment by abangser** [comment]
Thanks both for the reviews! So I think a few things...

1. The "main point" of this PR is to intro the design of how a tag will show up on the page. That hasn't had a lot of pushback since Chris so nicely improved on my ideas so I suspect those are OK?
2. The next most important thing is that we have a happy set up of starting tags. I agree with what you said Josh that we have "announcement" and "article" and then we can have tag of "community contribution". Let's keep it simple and add as we go.
3. There is an idea to formalise this. I think we can add these tags to the contribution guidelines / template for blog submissions as a more "manual" tracking of this. As for automated I am not sure there is a mechanism for that so will leave it to Hugo experts!

**Comment by joshgav** [comment]
@abangser agreed with all your points - once we're okay with the categories and tags I think you should merge, this is a good enhancement!

Also agree with documenting the tags in use somewhere as a "minimum viable" process. Maybe a Hugo expert will come along with a better idea eventually :smile:.

**Review by joshgav — APPROVED** [review]
LGTM when you're ready to merge

**Comment by abangser** [comment]
Per the conversation here I am going to merge this PR with only the simple categorisation and tagging in the markdown files. This will have no visible impact for now but will be set up for testing #522 as that get completed next month!

**Pull request merged**
